src/script/lua_api/l_item.cpp

```diff
@@ -525,6 +525,29 @@ int ModApiItemMod::l_register_item_raw(lua_State *L)
 	return 0; /* number of results */
 }
 
+// unregister_item(name)
+int ModApiItemMod::l_unregister_item_raw(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	std::string name = luaL_checkstring(L, 1);
+
+	// Get the writable item definition manager from the server
+	IWritableItemDefManager *idef =
+			getServer(L)->getWritableItemDefManager();
+
+	// Unregister the node
+	if (idef->get(name).type == ITEM_NODE) {
+		IWritableNodeDefManager *ndef =
```

Comment:
No, I mean the `IWritableNodeDefManager *ndef = ...`
If you move it outside of the loop then you only call getNodeDefManager() once, and reuse the same nodedef throughout the operation.  Remember, an IGameDef is an abstract interface, and maybe in the future getNodeDefManager might not just return a private member variable.  What if it does a heavier operation?  What if a new IGameDef implementation implements getNodeDefManager in a different quirkier way?